tau.py

```diff
@@ -30,3 +35,18 @@ def getES(self, df, source_dict):
                 df = df.Define(f'Tau_p4_{syst_name}_delta', f'Tau_p4_{syst_name} - Tau_p4_{nano}')
 
         return df, source_dict
+
+    def getSF(self, df, source_dict, deepTauVersion):
+        for source in [ central ] + TauCorrProducer.SFSources_dm+ TauCorrProducer.SFSources_pt:
+            updateSourceDict(source_dict, source, 'Tau')
+            genuineTau_SFtype = 'dm' if source in TauCorrProducer.SFSources_dm else 'pt'
+            for scale in getScales(source):
+                syst_name = getSystName(source, scale)
+                df = df.Define(f'Tau_p4_{syst_name}', f'''::correction::TauCorrProvider::getGlobal().getSF(
```

Comment:
- WPs and deepTau version should be defined in the config and WPs depend on the channel
- apply only to the HTT legs
- store both pt and dm SFs
- keep separate lists for each source type